File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] indexes = new int[2];
        int left = 0;
        int right = numbers.length - 1;
        while (left < right) {
            int sum = numbers[left] + numbers[right];
            if (sum == target) {
                indexes[0] = left + 1;
                indexes[1] = right + 1;
                return indexes;
            } else if (sum < target) {
                left++;
            } else {
                right--;
            }
        }
        return null;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Use hash map to keep track of number indexes
        indexes = {}
        for i, nr in enumerate(nums):
            looking_for = target - nr
            if looking_for in indexes:
                return [indexes[looking_for], i]
            indexes[nr] = i